controlplane/test/label.test.ts

```diff
@@ -248,4 +248,299 @@ describe('Labels', (ctx) => {
+    expect(graph2AfterUnset.subgraphs.length).toBe(2);
+
+    await server.close();
+  });
+
+  test('Unset federated graph label matchers', async (testContext) => {
+    const { client, server } = await SetupTest({ dbname });
+
+    const fedGraph1Name = genID('fedGraph1');
+    const fedGraph2Name = genID('fedGraph2');
+    const subgraph1Name = genID('subgraph1');
+    const subgraph2Name = genID('subgraph2');
+    const label1 = genUniqueLabel('label1');
+    const label2 = genUniqueLabel('label2');
+
+    await createFederatedGraph(client, fedGraph1Name, 'default', [joinLabel(label1)], 'http://localhost:8081');
+    await createFederatedGraph(client, fedGraph2Name, 'default', [], 'http://localhost:8082');
+
+    const createSubgraph = async (name: string, labels: Label[], routingUrl: string) => {
+      const createRes = await client.createFederatedSubgraph({
+        name,
+        labels,
+        routingUrl,
+        namespace: 'default',
+      });
+      expect(createRes.response?.code).toBe(EnumStatusCode.OK);
+      const publishResp = await client.publishFederatedSubgraph({
+        name,
+        namespace: 'default',
+        schema: Uint8Array.from(Buffer.from(`type Query { name: String! }`)),
+      });
+      expect(publishResp.response?.code).toBe(EnumStatusCode.OK);
+    };
+
+    await createSubgraph(subgraph1Name, [label1], 'http://localhost:8083');
+    await createSubgraph(subgraph2Name, [], 'http://localhost:8084');
+
+    // fedGraph1 should have subgraph1 and fedGraph2 should have subgraph2
+    const graph1 = await client.getFederatedGraphByName({
+      name: fedGraph1Name,
+      namespace: 'default',
+    });
+    expect(graph1.response?.code).toBe(EnumStatusCode.OK);
+    expect(graph1.subgraphs.length).toBe(1);
+    expect(graph1.subgraphs[0].name).toBe(subgraph1Name);
+
+    const graph2 = await client.getFederatedGraphByName({
+      name: fedGraph2Name,
+      namespace: 'default',
+    });
+    expect(graph2.response?.code).toBe(EnumStatusCode.OK);
+    expect(graph2.subgraphs.length).toBe(1);
+    expect(graph2.subgraphs[0].name).toBe(subgraph2Name);
+
+    await client.updateFederatedGraph({
+      name: fedGraph1Name,
+      namespace: 'default',
+      unsetLabelMatchers: true,
+      labelMatchers: [joinLabel(label2)], // We pass this to make sure that new label matchers are not created when unsetting
+    });
+
+    // fedGraph1 should have subgraph2
+    const graph1AfterUnset = await client.getFederatedGraphByName({
+      name: fedGraph1Name,
+      namespace: 'default',
+    });
+    expect(graph1AfterUnset.response?.code).toBe(EnumStatusCode.OK);
+    expect(graph1AfterUnset.graph?.labelMatchers.length).toBe(0);
+    expect(graph1AfterUnset.subgraphs.length).toBe(1);
+    expect(graph1AfterUnset.subgraphs[0].name).toBe(subgraph2Name);
+
+    await server.close();
+  });
+
+  test('Unset and set subgraph label', async (testContext) => {
```

Comment:
Could you be a bit more descriptive here? What use cases does it test? I see the following cases:

A User creates a fed graph or subgraph first with empty labels/matchers
A User adds labels/matchers afterwards
A User unsets labels/matchers from an existing graph

Reply:
It is difficult to describe user flow in a sentence. I have added comments. I think I have covered all cases

Reply:
You can try to describe the test expectation e.g. `Should compose correctly when removing all labels from a subgraph.` If you end up in a very long sentences it is a good sign to split up your test.

Reply:
Improved them